crux-lucene/src/crux/lucene.clj

```diff
@@ -70,6 +69,7 @@
   (user-data->tx-id (into {} (.getLiveCommitData index-writer))))
 
 (defn- search* [{:keys [^SearcherManager searcher-manager]}, ^Query q]
+  (.maybeRefreshBlocking searcher-manager)
```

Comment:
as discussed, let's err on the side of consistency here - we can consider a catching-up mode or similar.